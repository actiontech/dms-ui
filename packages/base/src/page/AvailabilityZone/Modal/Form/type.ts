import { FormInstance } from 'antd';

export type AvailabilityZoneFormType = {
  name: string;
  address: string;
};

export type AvailabilityZoneFormProps = {
  form: FormInstance<AvailabilityZoneFormType>;
  isUpdate?: boolean;
};
