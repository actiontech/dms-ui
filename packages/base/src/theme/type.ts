import { WorkflowRecordStatusEnum } from '@actiontech/shared/lib/api/base/service/common.enum';

export type SideMenuTheme = {
  suffixIconColor: string;
  projectSelector: {
    dropdown: {
      labelColor: string;
      iconColor: string;
      activeIconColor: string;
      groupLabelColor: string;
    };
  };
};

export type GuidanceTheme = {
  padding: number;
  gap: number;
  sceneSegmented: {
    backgroundColor: string;
    borderBottom: string;
  };
  steps: {
    icon: {
      width: number;
      height: number;
      backgroundColor: string;
    };
    connection: {
      width: number;
      gap: number;
      backgroundColor: string;
    };
    step: {
      border: string;
      title: {
        borderRight: string;
        index: {
          color: string;
        };
        text: {
          color: string;
        };
      };
      content: {
        desc: {
          color: string;
        };
      };
    };
  };
  guidanceButton: {
    default: {
      background: string;
      color: string;
    };
    hover: {
      background: string;
      color: string;
    };
  };
};

export type SystemTheme = {
  configButton: {
    backgroundColor: string;
    border: string;
  };
  logo: {
    boxShadow: string;
  };
};

export type DataExportTheme = {
  create: {
    form: {
      baseInfoTitleIconColor: string;
    };
  };
  statistics: {
    auditResultStatusColor: {
      [key in WorkflowRecordStatusEnum]: string;
    };
  };
};

export type IconTheme = {
  bindUser: {
    user: string;
    password: string;
  };
  dataExport: {
    infoCircle: string;
    clock: string;
  };
  home: {
    common: string;
  };
  system: {
    basicTitleTips: string;
    modify: string;
  };
};
