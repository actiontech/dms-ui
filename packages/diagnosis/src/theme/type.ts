export type MonitorSourceConfigTheme = {
  headerWrapper: {
    borderBottom: string;
    title: {
      fontSize: string;
      fontWeight: number;
      color: string;
    };
    refreshIcon: {
      color: string;
    };
    tagWrapper: {
      fontSize: string;
      fontWeight: number;
      color: string;

      tagItem: {
        border: string;
        backgroundColor: string;
      };
      tagPrimary: {
        primaryColor: string;
        hoverColor: string;
      };
      icon: {
        backgroundColor: string;
      };
    };
  };
};

export type LoginTheme = {
  logo: {
    boxShadow: string;
  };
  icon: {
    user: string;
    password: string;
  };
};

export type UserManagementTheme = {
  permissionType: {
    border: string;
    backgroundColor: string;
    color: string;
    activeBackgroundColor: string;
    activeColor: string;
  };
};
