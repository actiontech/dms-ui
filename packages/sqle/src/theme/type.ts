import { WorkflowRecordResV2StatusEnum } from '@actiontech/shared/lib/api/sqle/service/common.enum';

export type ExecWorkflowTheme = {
  list: {
    desc: {
      copyIconColor: string;
      hoverCopyIconBackgroundColor: string;
    };
  };
  create: {
    auditResult: {
      auditResultDrawer: {
        numberColor: string;
      };
      toggleButton: {
        borderColor: string;
        color: string;
        activeColor: string;
        bgColor: string;
        activeBgColor: string;
      };
      download: {
        borderColor: string;
        bgColor: string;
        boxShadow: string;
        itemColor: string;
        itemHoverColor: string;
        itemIconColor: string;
      };
    };
    form: {
      baseInfoTitleIconColor: string;
    };
    editForm: {
      titleColor: string;
      projectFlagIconColor: string;
    };
  };
  steps: {
    boxShadow: string;
  };
  common: {
    basicInfo: {
      borderColor: string;
      bgColor: string;
      titleColor: string;
      descColor: string;
    };
    auditResultFilter: {
      borderColor: string;
      bgColor: string;
      auditResultInfo: {
        itemBgColor: string;
        textColor: string;
        schemaValueColor: string;
      };
      options: {
        bgColor: string;
        activeBgColor: string;

        textColor: string;
        textActiveColor: string;
        numColor: string;
        numActiveColor: string;
      };
    };
  };
};

export type ReportStatisticsTheme = {
  bgColor: string;
  loadingColor: string;
  cardShow: {
    titleColor: string;
    iconColor: string;
    numberContColor: string;
    noteColor: string;
    contentFontSize: number;
    contentFontWeight: number;
  };
  CardWrapper: {
    titleColor: string;
    titleBorderColor: string;
  };
  TableTopList: {
    titleColor: string;
    noteTipColor: string;
    colorFillQuaternary: string;
    bgColor: {
      toColor: string;
      line1: string;
      line2: string;
      line3: string;
    };
  };
  ChartContTitle: {
    mainColor: string;
    mainSubColor: string;
    subContColor: string;
  };
  DatabaseTypeOrder: {
    default: {
      fontSize: number;
    };
    tooltip: {
      background: string;
      border: string;
      boxShadow: string;
    };
  };
  DatabaseSourceOrder: {
    default: {
      colorText: string;
      colorTextTertiary: string;
    };
  };
  LicenseStatistics: {
    ChartContTitle: {
      lineColor: string;
    };
    LicenseColumn: {
      defaultColor: string;
      fillColor: {
        column: string;
        columnBackground: string;
        xAxis: string;
        yAxis: string;
        state: {
          active: string;
        };
      };
    };
  };
  WorkOrderState: {
    ChartContTitle: {
      color: string;
      noteSubColor: string;
    };
  };
};

export type StatisticsTheme = {
  rectColor: string[];
  rectColorName: { [key: string]: string };
  auditRateStatus: {
    success: {
      color: string;
      bg: string;
    };
    warning: {
      color: string;
      bg: string;
    };
    error: {
      color: string;
      bg: string;
    };
    tip: {
      color: string;
      bg: string;
    };
  };
  auditResultStatusColor: {
    [key in WorkflowRecordResV2StatusEnum]: string;
  };
};

export type ProjectOverviewTheme = {
  SqlCount: {
    baseColor: string;
    grayColor: string;
  };
  DataSourceCount: {
    health: string;
    risk: string;
  };
  ProjectScore: {
    level: {
      dangerous: string;
      warning: string;
      good: string;
      excellent: string;
    };
    indicator: {
      pointer: string;
      pin: {
        fill: string;
        stroke: string;
      };
    };
  };
  ScanTask: {
    bar: {
      fill: string;
      bg: string;
      activeColor: string;
      label: {
        fill: string;
      };
      toolTip: {
        dotColor: string;
      };
    };
    detail: {
      stroke: string;
    };
  };
  DataSourcePerformance: {
    bar: {
      fill: string;
      bg: string;
    };
    toolTip: {
      dotColor: string;
    };
  };
};

export type WorkflowTemplateTheme = {
  progress: {
    normal: string;
    notice: string;
    warning: string;
    error: string;
    remainColor: string;
  };
  step: {
    honour: string;
    addFile: string;
    audit: string;
    sendPlane: string;
  };
  stepCard: {
    userAvatarBorder: string;
    boxShadow: string;
    hoverBoxShadow: string;
    backgroundColor: string;
    disableBorder: string;
    disabledBackgroundColor: string;
    disableHoverBackgroundColor: string;
    cardDesc: string;
    cardOperator: {
      borderTop: string;
      titleColor: string;
    };
    stepLineColor: string;
    stepReviewIndexTextColor: string;
    closeButtonBoxShadow: string;
    closeButtonFontSize: string;
    cardTitleFontWeight: number;
    descFontSize: string;
    operatorTitleFontWeight: number;
    stepReviewIndexTextFontWeight: number;
    authTextColor: string;
  };
  updateWorkflowTemplateStepInfo: {
    titleFontSize: string;
    titleFontWeight: number;
    infoFontSize: string;
    title: {
      titleWrapperBorderBottom: string;
      titleColor: string;
      titleInfoColor: string;
    };
    info: {
      titleColor: string;
    };
    stepInfoWrapper: {
      border: string;
      backgroundColor: string;
      textColor: string;
    };
    stepNodeAlert: {
      backgroundColor: string;
      alertTitleColor: string;
      alertContentColor: string;
    };
  };
  workflowTemplateAuthInfo: {
    topLevelFontWeight: number;
    authLevelFontWeight: number;
    authLevelFontSize: string;
    authInfoFontSize: string;
    borderBottom: string;
    authInfoColor: string;
  };
};

export type IconTheme = {
  execWorkFlow: {
    minusCircleFilledDisabled: string;
    profileSquareFilled: string;
    databaseFilled: string;
    schemaFilled: string;
    fileList: string;
    clock: string;
  };
  workflowTemplate: {
    common: string;
    userCircleFilled: string;
  };
};

export type DataSourceComparisonTheme = {
  comparisonResultDiffBackgroundColor: string;
  comparisonEntry: {
    card: {
      boxShadow: string;
      borderColor: string;
    };
    timeInfo: {
      color: string;
    };
    overview: {
      card: {
        hoverBoxShadow: string;
        suggestion: {
          color: string;
        };
        mode: {
          same: {
            borderColor: string;
            color: string;
            selectedBgColor: string;
          };
          inconsistent: {
            borderColor: string;
            color: string;
            selectedBgColor: string;
          };
          baseNotExist: {
            borderColor: string;
            color: string;
            selectedBgColor: string;
          };
          comparisonNotExist: {
            borderColor: string;
            color: string;
            selectedBgColor: string;
          };
        };
      };
    };
    treeNode: {
      title: {
        bgColor: string;
      };
      node: {
        inconsistent: {
          color: string;
          baselineBgColor: string;
          comparisonBgColor: string;
        };
        missing: {
          color: string;
          baselineBgColor: string;
          comparisonBgColor: string;
        };
        new: {
          color: string;
          bgColor: string;
        };
        tag: {
          inconsistent: {
            color: string;
            bgColor: string;
          };
          missing: {
            color: string;
            bgColor: string;
          };
          new: {
            color: string;
            bgColor: string;
          };
        };
      };
    };
  };
};

export type KnowledgeTheme = {
  highlight: {
    color: string;
    background: string;
  };
  result: {
    list: {
      boxShadow: string;
      hoverBoxShadow: string;
    };
    description: {
      borderLeftColor: string;
    };
  };
  graph: {
    wrapper: {
      borderColor: string;
      backgroundColor: string;
      dotColor: string;
    };
    container: {
      fullScreenBgColor: string;
    };
    control: {
      borderColor: string;
    };
    edge: {
      color: string;
    };
  };
};
