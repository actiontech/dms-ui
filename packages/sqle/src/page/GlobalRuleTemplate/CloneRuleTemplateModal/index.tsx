import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import { Col, Form, Row, Space, Typography } from 'antd';
import { BasicButton, BasicDrawer, BasicInput } from '@actiontech/shared';
import { formItemLayout } from '@actiontech/shared/lib/components/FormCom/style';
import { nameRule } from '@actiontech/shared/lib/utils/FormRule';
import { CloneRuleTemplateModalProps } from './index.type';

const CloneRuleTemplateModal: React.FC<CloneRuleTemplateModalProps> = ({
  visible,
  onClose,
  loading,
  onSubmit,
  link,
  templateName,
  form
}) => {
  const { t } = useTranslation();

  return (
    <BasicDrawer
      open={visible}
      title={t('ruleTemplate.cloneRuleTemplate.title')}
      closable={false}
      onClose={onClose}
      footer={
        <Space>
          <BasicButton onClick={onClose} disabled={loading}>
            {t('common.close')}
          </BasicButton>
          <BasicButton type="primary" onClick={onSubmit} loading={loading}>
            {t('common.submit')}
          </BasicButton>
        </Space>
      }
    >
      <Space direction="vertical" className="full-width-element">
        <Row>
          <Col>
            {t('ruleTemplate.cloneRuleTemplate.currentTemplateTips')}
            <Link target="_blank" to={link}>
              {templateName}
            </Link>
          </Col>
          <Col>
            <Typography.Text type="secondary">
              {t('ruleTemplate.cloneRuleTemplate.cloneDesc')}
            </Typography.Text>
          </Col>
        </Row>
        <Form form={form} {...formItemLayout.fullLine}>
          <Form.Item
            label={t('ruleTemplate.ruleTemplateForm.templateName')}
            name="templateName"
            validateFirst={true}
            rules={[
              {
                required: true
              },
              ...nameRule()
            ]}
          >
            <BasicInput
              placeholder={t('common.form.placeholder.input', {
                name: t('ruleTemplate.ruleTemplateForm.templateName')
              })}
            />
          </Form.Item>
          <Form.Item
            label={t('ruleTemplate.ruleTemplateForm.templateDesc')}
            name="templateDesc"
          >
            <BasicInput.TextArea
              className="textarea-no-resize"
              placeholder={t('common.form.placeholder.input', {
                name: t('ruleTemplate.ruleTemplateForm.templateDesc')
              })}
            />
          </Form.Item>
        </Form>
      </Space>
    </BasicDrawer>
  );
};

export default CloneRuleTemplateModal;
