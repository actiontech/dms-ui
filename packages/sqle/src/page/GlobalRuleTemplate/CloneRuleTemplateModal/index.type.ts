import { FormInstance } from 'antd';

export type CloneRuleTemplateFormFields = {
  templateName: string;
  templateDesc?: string;
};

export type CloneRuleTemplateModalProps = {
  link: string;
  templateName: string | React.ReactNode;
  loading: boolean;
  visible: boolean;
  form: FormInstance<CloneRuleTemplateFormFields>;
  onClose: () => void;
  onSubmit: () => void;
};
