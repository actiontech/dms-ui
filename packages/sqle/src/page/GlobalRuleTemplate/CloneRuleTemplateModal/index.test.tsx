import {
  renderWithReduxAndTheme,
  renderHooksWithTheme
} from '@actiontech/shared/lib/testUtil/customRender';
import { Form } from 'antd';
import CloneRuleTemplateModal from '../CloneRuleTemplateModal';
import { CloneRuleTemplateFormFields } from './index.type';
import { BrowserRouter } from 'react-router-dom';

describe('sqle/GlobalRuleTemplate/CloneRuleTemplateModal', () => {
  it('should match snapshot', () => {
    const { result } = renderHooksWithTheme(() =>
      Form.useForm<CloneRuleTemplateFormFields>()
    );
    const { baseElement } = renderWithReduxAndTheme(
      <BrowserRouter>
        <CloneRuleTemplateModal
          form={result.current[0]}
          visible={true}
          loading={false}
          link=""
          templateName="test_template"
          onClose={jest.fn()}
          onSubmit={jest.fn()}
        />
      </BrowserRouter>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
