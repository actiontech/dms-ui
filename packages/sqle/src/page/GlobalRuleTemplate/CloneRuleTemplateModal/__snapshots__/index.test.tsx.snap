// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sqle/GlobalRuleTemplate/CloneRuleTemplateModal should match snapshot 1`] = `
<body>
  <div />
  <div />
  <div>
    <div
      class="ant-drawer ant-drawer-right ant-drawer-open"
      tabindex="-1"
    >
      <div
        class="ant-drawer-mask"
      />
      <div
        aria-hidden="true"
        data-sentinel="start"
        style="width: 0px; height: 0px; overflow: hidden; outline: none; position: absolute;"
        tabindex="0"
      />
      <div
        class="ant-drawer-content-wrapper"
        style="width: 480px;"
      >
        <div
          aria-modal="true"
          class="ant-drawer-content basic-drawer-wrapper css-1q29jje"
          role="dialog"
        >
          <div
            class="ant-drawer-wrapper-body"
          >
            <div
              class="ant-drawer-header"
            >
              <div
                class="ant-drawer-header-title"
              >
                <div
                  class="ant-drawer-title"
                >
                  克隆规则模板
                </div>
              </div>
            </div>
            <div
              class="ant-drawer-body"
            >
              <div
                class="closed-icon-custom"
              >
                <span
                  aria-label="close"
                  class="anticon anticon-close"
                  data-testid="basic-drawer-close-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="close"
                    fill="currentColor"
                    fill-rule="evenodd"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="ant-space ant-space-vertical full-width-element"
              >
                <div
                  class="ant-space-item"
                  style="margin-bottom: 8px;"
                >
                  <div
                    class="ant-row"
                  >
                    <div
                      class="ant-col"
                    >
                      正在克隆审核规则模板
                      <a
                        href="/"
                        target="_blank"
                      >
                        test_template
                      </a>
                    </div>
                    <div
                      class="ant-col"
                    >
                      <span
                        class="ant-typography ant-typography-secondary"
                      >
                        克隆的规则模板只会继承源模板所有启用的规则、以及变更过的规则等级和阈值。克隆出的新规则模板的模板名称等基本信息需要手动填写。
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-space-item"
                >
                  <form
                    class="ant-form ant-form-horizontal"
                  >
                    <div
                      class="ant-form-item"
                    >
                      <div
                        class="ant-row ant-form-item-row"
                      >
                        <div
                          class="ant-col ant-col-24 ant-form-item-label"
                        >
                          <label
                            class="ant-form-item-required"
                            for="templateName"
                            title="模板名称"
                          >
                            模板名称
                          </label>
                        </div>
                        <div
                          class="ant-col ant-col-24 ant-form-item-control"
                        >
                          <div
                            class="ant-form-item-control-input"
                          >
                            <div
                              class="ant-form-item-control-input-content"
                            >
                              <input
                                aria-required="true"
                                class="ant-input basic-input-wrapper css-1pd1cd5"
                                id="templateName"
                                placeholder="请输入模板名称"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-form-item"
                    >
                      <div
                        class="ant-row ant-form-item-row"
                      >
                        <div
                          class="ant-col ant-col-24 ant-form-item-label"
                        >
                          <label
                            class=""
                            for="templateDesc"
                            title="模板描述"
                          >
                            模板描述
                          </label>
                        </div>
                        <div
                          class="ant-col ant-col-24 ant-form-item-control"
                        >
                          <div
                            class="ant-form-item-control-input"
                          >
                            <div
                              class="ant-form-item-control-input-content"
                            >
                              <textarea
                                class="ant-input basic-input-wrapper textarea-no-resize css-1pd1cd5"
                                id="templateDesc"
                                placeholder="请输入模板描述"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </form>
                </div>
              </div>
            </div>
            <div
              class="ant-drawer-footer"
            >
              <div
                class="ant-space ant-space-horizontal ant-space-align-center"
              >
                <div
                  class="ant-space-item"
                  style="margin-right: 8px;"
                >
                  <button
                    class="ant-btn ant-btn-default basic-button-wrapper css-1uxy5k6"
                    type="button"
                  >
                    <span>
                      关 闭
                    </span>
                  </button>
                </div>
                <div
                  class="ant-space-item"
                >
                  <button
                    class="ant-btn ant-btn-primary basic-button-wrapper css-1uxy5k6"
                    type="button"
                  >
                    <span>
                      提 交
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-sentinel="end"
        style="width: 0px; height: 0px; overflow: hidden; outline: none; position: absolute;"
        tabindex="0"
      />
    </div>
  </div>
</body>
`;
